File: backend/auth-service/src/main/java/com/sudarshanchakra/auth/dto/AuthResponse.java
package com.sudarshanchakra.auth.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Authentication response with JWT tokens")
public class AuthResponse {

    @Schema(description = "JWT access token")
    private String token;

    @Schema(description = "JWT refresh token")
    private String refreshToken;

    @Schema(description = "User information")
    private UserResponse user;
}

File: android/app/src/main/java/com/sudarshanchakra/domain/model/AuthResponse.kt
package com.sudarshanchakra.domain.model

data class AuthResponse(
    val token: String,
    val refreshToken: String,
    val user: User
)

data class LoginRequest(
    val username: String,
    val password: String
)

data class RegisterRequest(
    val username: String,
    val email: String,
    val password: String
)
